include/miral/miral/display_configuration.h

```diff
@@ -173,8 +171,8 @@ class DisplayConfiguration
     ///
     /// The value of this attribute will be made available through
     /// #miral::UserDisplayConfigurationOutput::custom_attribute at the
-    /// \p key. Note that the data is value for the key has to have a
-    /// type of `std::optional<std::string>`.
+    /// \p key. Note that the value at the \p key must have a type of
+    /// `std::optional<std::string>`.
```

Comment:
I can guess, but the language is entirely unclear what the "Note that the value at the `key` must have a type of `std::optional<std::string>`" requirement means to the calling code.

If the reference is to types in the `.yaml`, then use of C++ types is inappropriate